File: lib/models/product.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class Product {
  final String name;
  final String description;
  final double quantity;
  final String category;
  final double price;
  final String? id;
  final List<String> images;

  Product({
    required this.name,
    required this.description,
    required this.quantity,
    required this.images,
    required this.category,
    required this.price,
    this.id,
   
  });
  // rating

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'description': description,
      'quantity': quantity,
      'category': category,
      'price': price,
      'id': id,
      'images': images,
    };
  }

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
        name: map['name'] as String,
        description: map['description'] as String,
        quantity: map['quantity'] as double,
        category: map['category'] as String,
        price: map['price'] as double,
        id: map['_id'] != null ? map['id'] as String : null,
        images: List<String>.from(
          (map['images'] as List<String>),
        ));
  }

  String toJson() => json.encode(toMap());

  factory Product.fromJson(String source) =>
      Product.fromMap(json.decode(source) as Map<String, dynamic>);
}

File: server/models/product.js
const mongooes = require('mongoose');

const productSchema = mongooes.Schema({
    name: {
        trim: true,
        type: String,
        required: true,
    },
    description: {
        trim: true,
        type: String,
        required: true,
    },
    // way to store array
    images: [{
        type: String,
        required: true,
    }],
    price: {
        type: Number,
        required: true,
    }, 
    quantity: {
        type: Number,
        required: true,
    }, 
    category: {
        trim: true,
        type: String,
        required: true,
    },
    // rating
});

const Product = mongooes.model("Product", productSchema);
module.exports = Product